Add custom server certs installation with foremanctl (#4821)
theforeman/foreman-documentation: guides/common/modules/proc_deploying-a-custom-ssl-certificate-to-foreman-server.adoc

[IMPORTANT]
====
Do not store the SSL certificates or `.tar` bundles in `/tmp` or `/var/tmp` directory.
The operating system removes files from these directories periodically.
As a result,
ifdef::foremanctl[`{foremanctl-installer}`]
ifndef::foremanctl[`{foreman-installer}`]
fails to execute while enabling features or upgrading {ProjectServer}.
====

.Procedure
* Update certificates on your {ProjectServer}:
+
[options="nowrap", subs="+quotes,attributes"]
ifdef::foremanctl[]
----
# {foremanctl-installer} deploy \

> --certificate-source=custom_server \

Reviewer: Just to keep it consistent
```suggestion
--certificate-source custom_server \
```

Reviewer 2: This made me thing (outside this particular PR): is there a preferred form for this in the style guide and can VALE enforce this for us?

Author: @evgeni The preferred form is definitely without the equals sign, but it's not codified anywhere (RH SSG or IBM SG or Contributing).

Author: I left some of those in... fixing in https://github.com/theforeman/foreman-documentation/pull/4840

--certificate-server-certificate "_/root/{project-context}_cert/{project-context}_cert.pem_" \
--certificate-server-key "_/root/{project-context}_cert/{project-context}_cert_key.pem_" \
--certificate-server-ca-certificate "_/root/{project-context}_cert/ca_cert_bundle.pem_"
----
+
The options used in the command include the following:
+

> `--certificate-source=custom_server`:: Sets the certificate source to custom server certificates provided by the user. Automatically generates an internal CA for client certificates and localhost.

Reviewer 2: I don't think the second sentence is important to the user

Author: ```suggestion
`--certificate-source=custom_server`:: Sets the certificate source to custom server certificates provided by the user.
```

Reviewer 3: I think it sets the expectations correctly. I would not remove that.

`--certificate-server-certificate`:: Path to {ProjectServer} certificate file that is signed by a Certificate Authority.

> `--certificate-server-key`:: Path to the private key that was used to sign {ProjectServer} certificate.

Reviewer 2: the key is the one for the cert, not for the signing of the cert :)

Author: Is this correct?

```suggestion
`--certificate-server-key`:: Path to the private key for the {ProjectServer} certificate.
```

Reviewer 2: yes

> `--certs-server-ca-cert`:: Path to the Certificate Authority bundle.

Reviewer 3: ```suggestion
`--certificate-server-certificate`:: Path to {ProjectServer} certificate file that is signed by a Certificate Authority.
`--certificate-server-key`:: Path to the private key that was used to sign {ProjectServer} certificate.
`--certificate-server-ca-certificate`:: Path to the Certificate Authority bundle.
```
According to https://github.com/theforeman/foremanctl/pull/462/changes#diff-37d6211d09af130be7d66a0d0a0b9b99f93dbee347e5b0f66ee6c44e6ac40847

====
Do not store the SSL certificates or .tar bundles in `/tmp` or `/var/tmp` directory.
The operating system removes files from these directories periodically.
As a result,
ifdef::foremanctl[`{foremanctl-installer}`]
ifndef::foremanctl[`{foreman-installer}`]
fails to execute while enabling features or upgrading {ProjectServer}.
====

.Procedure
* Update certificates on your {ProjectServer}:
+
[options="nowrap", subs="+quotes,attributes"]
ifdef::foremanctl[]
----
# {foremanctl-installer} deploy \
--certificate-source=custom_server \
--certificate-server-certificate "_/root/{project-context}_cert/{project-context}_cert.pem_" \
--certificate-server-key "_/root/{project-context}_cert/{project-context}_cert_key.pem_" \
--certificate-server-ca-certificate "_/root/{project-context}_cert/ca_cert_bundle.pem_"
//--certs-update-server --certs-update-server-ca

> //Are these^ just unnecessary now?

Author: @ShimShtein Are these params unnecessary now? I don't see their equivalents in foremanctl.

Reviewer 3: Yes, they are not needed

----
+
The options used in the command include the following:
+

> `--certificate-source=custom_server`:: Sets the certificate source to custom server certificates provided by the user. Automatically generates an internal CA for client certificates and localhost. Server certificate, key, and CA bundle are copied to `/root/certificates/`.

Reviewer 3: I don't think we need to mention the "resting place" of the certificate files here.
a. we gonna change it
b. it's an internal folder, the user does not need to be aware of it

+
The options used in the command include the following:
+
`--certs-server-cert`:: Path to {ProjectServer} certificate file that is signed by a Certificate Authority.

> `--certs-server-key`:: Path to the private key that was used to sign {ProjectServer} certificate.

Reviewer 2: funny, that it's wrong here too

Author: ```suggestion
`--certs-server-key`:: Path to the private key for the {ProjectServer} certificate.
```

> `--certs-server-ca-cert`:: Path to the Certificate Authority bundle.

Reviewer 3: Ditto, according to https://github.com/theforeman/foremanctl/pull/462/changes#diff-37d6211d09af130be7d66a0d0a0b9b99f93dbee347e5b0f66ee6c44e6ac40847

Author: These are the non-containerized options which didn't change, or at least, we don't want to change them with this PR.

Reviewer 3: Yup, sorry, misread the `ifndef` statements.
